### PR TITLE
added stack class

### PR DESCRIPTION
**Pull request by marcelluethi**

### CODE REVIEW

File: Queue.java [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
+
+		void put(int x) {
+				if ((tail + 1) % size == head) {
+						overflow = true;
+				} else {
+						overflow = true;
+				}
+
+				if (overflow == true) {
+					q[tail]  =x;
```
Comment (paultroeger) [inline Queue.java:24]: Code not properly formatted

File: Queue.java [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
+
+		boolean underflow = false;
+		boolean overflow = false;
+
+		Queue(int size) {
+				this.size = size;
+				q = new int[size];
+		}
+
+		void put(int x) {
```
Comment (paultroeger) [inline Queue.java:16]: Better naming: x -> newQueueItem

File: Queue.java [hunk 3]
```diff
@@ -1,40 +1,40 @@
...
+				if ((tail + 1) % size == head) {
+						overflow = true;
+				} else {
+						overflow = true;
+				}
```
Comment (paultroeger) [inline Queue.java:21]: Unnecessary else statement. Wrong Logic.

File: Queue.java [hunk 4]
```diff
@@ -1,40 +1,40 @@
...
+
+				if (overflow == true) {
```
Comment (paultroeger) [inline Queue.java:23]: if (overflow)

File: Queue.java [hunk 5]
```diff
@@ -1,40 +1,40 @@
...
+					q[tail]  =x;
+					tail = (++tail) % size;
+				}
+		}
+
+		void get() {
+				underflow = head = tail;
```
Comment (paultroeger) [inline Queue.java:30]: Incorrect java programming

File: Queue.java [hunk 6]
```diff
@@ -1,40 +1,40 @@
...
+						overflow = true;
+				}
+
+				if (overflow == true) {
+					q[tail]  =x;
+					tail = (++tail) % size;
+				}
+		}
+
+		void get() {
```
Comment (paultroeger) [inline Queue.java:29]: return value should be int

File: Queue.java [hunk 7]
```diff
@@ -1,40 +1,40 @@
...
+				underflow = head = tail;
+				if (underflow) return -1;
```
Comment (paultroeger) [inline Queue.java:31]: Style inconsistent

File: Queue.java [hunk 8]
```diff
@@ -1,40 +1,40 @@
 class Queue {
```
Comment (paultroeger) [inline Queue.java:2]: No Comments and no Documentation

File: Stack.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    boolean underflow;
+
+    Stack(int size) {
+        s = new int[size];
+        length = 0;
+        overflow = false;
+        underflow = false;
+    }
+
+    void push(int x)   {
```
Comment (paultroeger) [inline Stack.java:16]: Variable not named correctly

File: Stack.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        if (length >= s.length) {
+            overflow = true;
+        }
+        else {
```
Comment (paultroeger) [inline Stack.java:20]: Inconsistent Style

File: Stack.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+            overflow = false;
+        }
+      if (overflow == true)  s[length++] = x;
```
Comment (paultroeger) [inline Stack.java:23]: if(overflow)

Inconsistent Style

File: Queue.java (hunk 2 above)
Comment (marcelluethi) [reply Queue.java:16]: Ich finde x viel besser, da es kürzer ist

**Pull request closed without merging**